apps/demo-game/src/reducers/SegmentReducer.ts

```diff
@@ -1,86 +1,67 @@
-import { Action } from '@gbl-uzh/platform'
+import { PayloadSegment } from '@gbl-uzh/platform'
 import {
   computeScenarioOutcome,
   debugLog,
   diceRoll,
 } from '@gbl-uzh/platform/dist/lib/util'
 import { PeriodFacts, PeriodSegmentFacts } from '@graphql/index'
-import { PrismaClient } from '@prisma/client'
+import { produce } from 'immer'
 import * as R from 'ramda'
-import { P, match } from 'ts-pattern'
 
-export enum ActionTypes {
-  SEGMENT_INITIALIZE = 'SEGMENT_INITIALIZE',
+type State = {
+  diceRolls?: { bonds: number; stocks: number }[]
+  returns?: { bank: number; bonds: number; stocks: number }[]
 }
 
-type Actions = Action<
-  ActionTypes.SEGMENT_INITIALIZE,
-  {
-    segmentIx: number
-    segmentCount: number
-    periodIx: number
-    periodFacts: PeriodFacts
-    previousSegmentFacts?: PeriodSegmentFacts
-  },
-  PrismaClient
->
+export function initialize(
+  state: State,
+  payload: PayloadSegment<PeriodFacts, PeriodSegmentFacts>
+) {
```

Comment:
we actually also want output states defined for all the functions. so in the output, diceRolls and returns has to be around. and in the input for initialize, we know that it is not around

Reply:
Yup, just waited for some feedback wrt to isDirty :)

Except from the action reducer, all should have a output type now in the demo game